keras_eval/data_generators.py

```diff
@@ -0,0 +1,328 @@
+                             target_size=None,
+                             interpolation=self.interpolation)
+
+            if self.crop_original == 'center_crop':
+                w, h = image.size
+                if w > h:
+                    image = image.crop((w / 2 - h / 2, 0, w / 2 + h / 2, h))
+                else:
+                    image = image.crop((0, h / 2 - w / 2, w, h / 2 + w / 2))
+            elif self.crop_original:
+                raise ValueError('crop_original entered not supported, only `center_crop` is being supported now')
+
+            image_w, image_h = image.size
+
+            if self.scale_sizes is not None:
+                for size in self.scale_sizes:
+                    if image_w <= image_h:
+                        img = image.resize((size, round(image_h / image_w * size)))
+                    else:
+                        img = image.resize((round(image_w / image_h * size), size))
+                    crops += self._apply_augmentation(img, self.target_size, self.transforms)
+
+            crops += [self._apply_transform(image.resize(self.target_size), transform) for transform in self.transforms]
+
+            for c_i, crop in enumerate(crops):
+                x = img_to_array(crop, data_format=self.data_format)
+                x = self.image_data_generator.standardize(x)
+                batch_x[i, c_i] = x
+
+        # optionally save augmented images to disk for debugging purposes
+        if self.save_to_dir:
+            for i, j in enumerate(index_array):
+                for crop in batch_x[i]:
+                    img = array_to_img(crop, self.data_format, scale=True)
+                    fname = '{prefix}_{index}_{hash}.{format}'.format(
+                        prefix=self.save_prefix,
+                        index=j,
+                        hash=np.random.randint(1e7),
+                        format=self.save_format)
+                    print(fname)
+                    img.save(os.path.join(self.save_to_dir, fname))
+
+        # build batch of labels
+        if self.class_mode == 'input':
+            batch_y = batch_x.copy()
+        elif self.class_mode == 'sparse':
+            batch_y = self.classes[index_array]
+        elif self.class_mode == 'binary':
+            batch_y = self.classes[index_array].astype(backend.floatx())
+        elif self.class_mode == 'categorical':
+            batch_y = np.zeros(
+                (len(batch_x), self.num_classes),
+                dtype=backend.floatx())
+            for i, label in enumerate(self.classes[index_array]):
+                batch_y[i, label] = 1.
+        else:
+            return batch_x
+        return batch_x, [batch_y for n in range(self.n_crops)]
+
+
+class AugmentedImageDataGenerator(ImageDataGenerator):
+    '''
+    ImageDataGenerator that returns multiple outputs using AugmentedDirectoryIterator as iterator.
+    '''
+
+    def __init__(self,
+                 data_augmentation,
```

Comment:
Shall we set it by default to `None`? 
```suggestion
                 data_augmentation=None,
```

Reply:
if we put None, then what is the use of class? 😄 
If `data_augmentation` is  None then we will perform the usual evaluation